### PR TITLE
Marketplace

### PR DESCRIPTION
**Pull request by Joeper214**
I branched out aws-marketplace from develop branch. and Created marketplace-WIP branch for code review.

### CODE REVIEW

File: app/assets/javascripts/app_settings.js [hunk 1]
```diff
@@ -11,6 +11,156 @@
...
+        log_directory: null,
+        access_key: null,
+        secret_access_key: null,
+        aws_region: null,
+        keypair_name: null,
+        keypair_value: null
+      },
+    },
+    methods: {
+      onFileChange(e) {
```
Comment (skyarch-networks) [inline app/assets/javascripts/app_settings.js]: __[jshint]__
*  'concise methods' is available in ES6 (use 'esversion

File: app/assets/javascripts/app_settings.js [hunk 2]
```diff
@@ -11,6 +11,156 @@
...
+          var files = e.target.files || e.dataTransfer.files;
+          if (!files.length) return;
+          this.createFile(files[0]);
+      },
+      createFile(file) {
```
Comment (skyarch-networks) [inline app/assets/javascripts/app_settings.js]: __[jshint]__
*  'concise methods' is available in ES6 (use 'esversion

File: app/assets/javascripts/app_settings.js [hunk 3]
```diff
@@ -11,6 +11,156 @@
...
+          var reader = new FileReader();
+          var vm = this.params;
+
+          reader.onload = (e) => {
```
Comment (skyarch-networks) [inline app/assets/javascripts/app_settings.js]: __[jshint]__
*  'arrow function syntax (=>)' is only available in ES6 (use 'esversion

File: app/assets/javascripts/app_settings.js [hunk 4]
```diff
@@ -11,6 +11,156 @@
...
+    },
+    computed: {
+      keysExists: function () {
+        var self = this;
+        return (self.params.access_key && self.params.secret_access_key && self.params.aws_region);
+      },
+      required_filed: function () {
+        var self = this;
+        return (self.params.access_key
+          && self.params.secret_access_key
```
Comment (skyarch-networks) [inline app/assets/javascripts/app_settings.js]: __[jshint]__
*  Bad line breaking before '&&'.

File: app/assets/javascripts/app_settings.js [hunk 5]
```diff
@@ -11,6 +11,156 @@
...
+          && self.params.aws_region
```
Comment (skyarch-networks) [inline app/assets/javascripts/app_settings.js]: __[jshint]__
*  Bad line breaking before '&&'.

File: app/assets/javascripts/app_settings.js [hunk 6]
```diff
@@ -11,6 +11,156 @@
...
+          && self.params.keypair_name
```
Comment (skyarch-networks) [inline app/assets/javascripts/app_settings.js]: __[jshint]__
*  Bad line breaking before '&&'.

File: app/assets/javascripts/app_settings.js [hunk 7]
```diff
@@ -11,6 +11,156 @@
...
+          && self.params.keypair_value
```
Comment (skyarch-networks) [inline app/assets/javascripts/app_settings.js]: __[jshint]__
*  Bad line breaking before '&&'.

**Comment by Joeper214** [comment]
Please review and Merge PR. @root-beer

**Comment by Joeper214** [comment]
Please use Amazon linux EC2 to test this feature. during fresh installation.

**Comment by Joeper214** [comment]
Next task is to create an Instructions for creating 

- [ ] client
- [ ] project  
- [ ] and infrastructure

**Comment by root-beer** [comment]
I'll review now.

**Comment by Joeper214** [comment]
よろしくおねがいします。

File: app/assets/javascripts/app_settings.js [hunk 8]
```diff
@@ -11,6 +11,156 @@
...
+      createFile: function(file) {
+          var reader = new FileReader();
+          var vm = this.params;
+
+          reader.onload = function (e) {
+              vm.keypair_value = e.target.result;
+          };
+          reader.readAsText(file);
+
+          if(!file.type.match('application/x-x509-ca-cert')){
```
Comment (Joeper214) [inline app/assets/javascripts/app_settings.js]: This is the filter. @root-beer

Comment (Joeper214) [inline app/assets/javascripts/app_settings.js]: Line `46`

Comment (Joeper214) [inline app/assets/javascripts/app_settings.js]: Maybe this is not safe to other browsers. I'll try to remove this.

Comment (root-beer) [inline app/assets/javascripts/app_settings.js]: お願いします 🙇

**Comment by root-beer** [comment]
LGTM. Thanks!

**Pull request merged**